controller/api/public/grpc_server.go

```diff
@@ -3,200 +3,55 @@ package public
-		var err error
-		labelSelector, err = labels.Parse(s)
-		if err != nil {
-			return nil, fmt.Errorf("invalid label selector \"%s\": %s", s, err)
-		}
-	}
-
-	nsQuery := ""
-	namespace := ""
-	if req.GetNamespace() != "" {
-		namespace = req.GetNamespace()
-	} else if targetOwner.GetNamespace() != "" {
-		namespace = targetOwner.GetNamespace()
-	} else if targetOwner.GetType() == pkgK8s.Namespace {
-		namespace = targetOwner.GetName()
-	}
-	if namespace != "" {
-		nsQuery = fmt.Sprintf("namespace=\"%s\"", namespace)
-	}
-	processStartTimeQuery := fmt.Sprintf(podQuery, nsQuery)
-
-	// Query Prometheus for all pods present
-	vec, err := s.queryProm(ctx, processStartTimeQuery)
-	if err != nil && !errors.Is(err, ErrNoPrometheusInstance) {
-		return nil, err
-	}
-
-	for _, sample := range vec {
-		pod := string(sample.Metric["pod"])
-		timestamp := sample.Timestamp
-
-		reports[pod] = podReport{
-			lastReport:              time.Unix(0, int64(timestamp)*int64(time.Millisecond)),
-			processStartTimeSeconds: time.Unix(0, int64(sample.Value)*int64(time.Second)),
-		}
-	}
-
-	var pods []*corev1.Pod
-	if namespace != "" {
-		pods, err = s.k8sAPI.Pod().Lister().Pods(namespace).List(labelSelector)
-	} else {
-		pods, err = s.k8sAPI.Pod().Lister().List(labelSelector)
-	}
-
-	if err != nil {
-		return nil, err
-	}
-	podList := make([]*pb.Pod, 0)
-
-	for _, pod := range pods {
-		if s.shouldIgnore(pod) {
-			continue
-		}
-
-		ownerKind, ownerName := s.k8sAPI.GetOwnerKindAndName(ctx, pod, false)
-		// filter out pods without matching owner
-		if targetOwner.GetNamespace() != "" && targetOwner.GetNamespace() != pod.GetNamespace() {
-			continue
-		}
-		if targetOwner.GetType() != "" && targetOwner.GetType() != ownerKind {
-			continue
-		}
-		if targetOwner.GetName() != "" && targetOwner.GetName() != ownerName {
-			continue
-		}
-
-		updated, added := reports[pod.Name]
-
-		item := util.K8sPodToPublicPod(*pod, ownerKind, ownerName)
-		item.Added = added
-
-		if added {
-			since := time.Since(updated.lastReport)
-			item.SinceLastReport = &duration.Duration{
-				Seconds: int64(since / time.Second),
-				Nanos:   int32(since % time.Second),
-			}
-			sinceStarting := time.Since(updated.processStartTimeSeconds)
-			item.Uptime = &duration.Duration{
-				Seconds: int64(sinceStarting / time.Second),
-				Nanos:   int32(sinceStarting % time.Second),
-			}
-		}
-
-		podList = append(podList, item)
-	}
-
-	rsp := pb.ListPodsResponse{Pods: podList}
-
-	log.Debugf("ListPods response: %s", rsp.String())
-
-	return &rsp, nil
+func (*grpcServer) Version(ctx context.Context, req *pb.Empty) (*pb.VersionInfo, error) {
+	return &pb.VersionInfo{GoVersion: runtime.Version(), ReleaseVersion: version.Version, BuildDate: "1970-01-01T00:00:00Z"}, nil
 }
 
-func (s *grpcServer) SelfCheck(ctx context.Context, in *healthcheckPb.SelfCheckRequest) (*healthcheckPb.SelfCheckResponse, error) {
+// TODO: Reenable this check just for checking the control plane can
```

Comment:
In order to make this work without taking a dependency on linkerd-viz, we'd need to introduce a new API or move SelfCheck to a shared location.  To be honest, I'm not sure if it's even worth it.  Especially since this process doesn't even need to talk to the k8s API.  Perhaps a better check would be to look for errors in the destination or identity controller logs or kubernetes event streams?

Reply:
But I think we can get rid of this TODO